scripts/nlp/evaluate.py

```diff
@@ -1,5 +1,86 @@
 # all the evaluation related function like: accuracy calc, false positive etc
+import spacy
+from spacy.gold import GoldParse
+from spacy.scorer import Scorer
+
+def evaluate(nlp, examples, ent='LOC_ACCIDENT'):
+    scorer = Scorer()
+    for input_, annot in examples:
+        text_entities = []
+        for entity in annot.get('entities'):
+            if ent in entity:
+                text_entities.append(entity)
+        doc_gold_text = nlp.make_doc(input_)
+        gold = GoldParse(doc_gold_text, entities=text_entities)
+        pred_value = nlp(input_)
+        scorer.score(pred_value, gold)
+    return scorer.scores
 
 
 def calculate_false_negatives(y_true, y_pred):
-    pass
\ No newline at end of file
+    pass
+
+def check_accuracy(test_list, model, test_set_name='PLACEHOLDER'):
+    correct = 0
+    wrong = 0
+
+    dict_phrase_issue = {}
+    dict_issue_location = {} 
+    dict_issue_not_found = {}
+    for i in range(len(test_list)):
+        ent_model_results = []
+        phrase_text = model(test_list[i][0])
+        entities_localtion = test_list[i][1]['entities']
+        for ent in phrase_text.ents:
+            tuple_ent =(ent.start_char, ent.end_char, ent.label_)
+            ent_model_results.append(tuple_ent)
+        if entities_localtion == ent_model_results:
+            correct += 1
+        else:
+            # False positive
+            wrong_guess = list(set(ent_model_results).difference(entities_localtion))
+            # False negative
+            didnt_find = list(set(entities_localtion).difference(ent_model_results))
+            
+            dict_phrase_issue[wrong] = test_list[i][0]
```

Comment:
why use dictionary instead of lists?

Reply:
A fost mai usor sa lucrez cu dict in cazul asta